### PR TITLE
Use `EnableDynamicLoading` for plugins

### PR DESCRIPTION
**Pull request by vitek-karas**
In .NET 6 SDK doesn't produce `.runtimeconfig.dev.json` by default and as such plugins with NuGet dependencies won't run with just `dotnet build` as they don't copy their dependencies to output.

Adding `EnableDynamicLoading=true` to a plugin project to fix this. Technically it's not necessary, and just `CopyLocalLockFileAssemblies=true` would be enough, but I think it's just really weird. Also it's much easier to explain: "I want to load this project dynamically at runtime, so I'm going to mark it with `EnableDynamicLoading=true`".

Modified the section which adressed this issue previous - mostly simplifying it.

### CODE REVIEW

**Comment by vitek-karas** [comment]
There's a similar change in the relevant sample here: https://github.com/dotnet/samples/pull/4773

Once that is approved I will copy the sample into the samples section in the docs repo as well.

**Comment by vitek-karas** [comment]
Once all of these are merge we can resolve https://github.com/dotnet/runtime/issues/57345.

File: docs/core/tutorials/creating-app-with-plugin-support.md [hunk 1]
```diff
@@ -244,6 +244,14 @@ Now, open the *HelloPlugin.csproj* file. It should look similar to the following
 
 ```
 
+In between the `<PropertyGroup>` tags, add the following element:
+
+```xml
+  <EnableDynamicLoading>true</EnableDynamicLoading>
+```
+
+The `<EnableDynamicLoading>true</EnableDynamicLoading>` prepares the project so that it can be used as a plugin. Among other things, this will copy all of its dependencies to the output of the project.
```
Comment (elinor-fung) [inline docs/core/tutorials/creating-app-with-plugin-support.md:253]: Maybe link back to `project-sdk/msbuild-props.md#enabledynamicloading`?

File: docs/core/tutorials/creating-app-with-plugin-support.md [hunk 2]
```diff
@@ -263,7 +271,7 @@ Now that the `HelloPlugin` project is complete, you should update the `AppWithPl
...
+Almost all plugins are more complex than a simple "Hello World", and many plugins have dependencies on other libraries. The `JsonPlugin` and `OldJsonPlugin` projects in the sample show two examples of plugins with NuGet package dependencies on `Newtonsoft.Json`. The project files themselves don't have any special information for the project references, and (after adding the plugin paths to the `pluginPaths` array) the plugins run perfectly, even if run in the same execution of the `AppWithPlugin` app. These projects added `<EnableDynamicLoading>true</EnableDynamicLoading>` to the project properties so that they copy all of their dependencies to the output of `dotnet build`. Publishing the class library with `dotnet publish` will also copy all of its dependencies to the publish output.
```
Comment (elinor-fung) [inline docs/core/tutorials/creating-app-with-plugin-support.md:274]: > The project files themselves don't have any special information for the project references, and (after adding the plugin paths to the `pluginPaths` array) the plugins run perfectly, even if run in the same execution of the `AppWithPlugin` app.

I'm not sure how useful this is anymore. It seemed more related to the dev vs user's machine line that was removed and looks out of place / more confusing than clarifying to me now.

>  These projects added `<EnableDynamicLoading>true</EnableDynamicLoading>`

Would it make sense to explicitly recommend that all plugins do this? (And if so, directly state that.)

**Review by elinor-fung — APPROVED** [review]

**Pull request merged**